src/helpers/Bmp.cpp

```diff
@@ -0,0 +1,126 @@
+#include "Bmp.hpp"
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+class BmpHeader {
+public:
+    unsigned char format[2];
+    uint32_t sizeOfFile;
+    uint16_t reserved1;
+    uint16_t reserved2;
+    uint32_t dataOffset;
+    uint32_t sizeOfBitmapHeader;
+    uint32_t width;
+    uint32_t height;
+    uint16_t numberOfColors;
+    uint16_t numberOfBitPerPixel;
+    uint32_t compressionMethod;
+    uint32_t imageSize;
+    uint32_t horizontalResolutionPPM;
+    uint32_t verticalResolutionPPM;
+    uint32_t numberOfCollors;
+    uint32_t numberOfImportantCollors;
+
+    BmpHeader(std::ifstream& file) {
+        file.seekg(0, std::ios::end);
+        uint32_t streamLength = file.tellg();
+        file.seekg(0, std::ios::beg);
+
+        file.read(reinterpret_cast<char*>(&format), sizeof(format));
+        if (!(format[0] == 66 && format[1] == 77)) {
+            Debug::log(ERR, "Unable to parse bitmap header: wrong bmp file type");
+            exit(1);
+        }
+
+        file.read(reinterpret_cast<char*>(&sizeOfFile), sizeof(sizeOfFile));
+
+        if (sizeOfFile != streamLength) {
+            Debug::log(ERR, "Unable to parse bitmap header: wrong value of file size header");
+            exit(1);
+        }
+
+        file.read(reinterpret_cast<char*>(&reserved1), sizeof(reserved1));
+        file.read(reinterpret_cast<char*>(&reserved2), sizeof(reserved2));
+        file.read(reinterpret_cast<char*>(&dataOffset), sizeof(dataOffset));
+        file.read(reinterpret_cast<char*>(&sizeOfBitmapHeader), sizeof(sizeOfBitmapHeader));
+        file.read(reinterpret_cast<char*>(&width), sizeof(width));
+        file.read(reinterpret_cast<char*>(&height), sizeof(height));
+        file.read(reinterpret_cast<char*>(&numberOfColors), sizeof(numberOfColors));
+        file.read(reinterpret_cast<char*>(&numberOfBitPerPixel), sizeof(numberOfBitPerPixel));
+        file.read(reinterpret_cast<char*>(&compressionMethod), sizeof(compressionMethod));
+        file.read(reinterpret_cast<char*>(&imageSize), sizeof(imageSize));
+        file.read(reinterpret_cast<char*>(&horizontalResolutionPPM), sizeof(horizontalResolutionPPM));
+        file.read(reinterpret_cast<char*>(&verticalResolutionPPM), sizeof(verticalResolutionPPM));
+        file.read(reinterpret_cast<char*>(&numberOfCollors), sizeof(numberOfCollors));
+        file.read(reinterpret_cast<char*>(&numberOfImportantCollors), sizeof(numberOfImportantCollors));
+
+        if (!imageSize)
+            imageSize = sizeOfFile - dataOffset;
+
+        if (imageSize != (width * height * numberOfBitPerPixel/8)) {
+            Debug::log(ERR, "Unable to parse bitmap header: wrong image size");
+            exit(1);
+        }
+
+        file.seekg(dataOffset);
+    };
+};
+
+void reflectImage(unsigned char* image, uint32_t numberOfRows, int stride) {
+    int rowStart = 0;
+    int rowEnd = numberOfRows - 1;
+    std::vector<unsigned char> temp;
+    temp.reserve(stride);
```

Comment:
.resize

Reply:
resize will initialize vector by zeroes, but there is no need of it, because the next action will copy the date to the vector. reserve will only allocate the memory i think that is enough here

Reply:
it will not update the vector size properly which I believe technically is ub